### PR TITLE
Actually Fix options for external cub and rocprim

### LINKED ISSUE #1194: RAJA does not respect my wish to not use an external CUB
https://github.com/LLNL/RAJA/blob/4a07809dfd0d8dcd4daf5eeb91fb692394c5815b/cmake/SetupPackages.cmake#L32

I don't want to use an external cub, but  no matter what I do, RAJA always uses the system install. I can't figure out if its even possible to use the submodule if there is a the is findable by `FindPackage`. Any advice?

### PR DESCRIPTION
**Pull request by MrBurmark**
# Summary
They now default to something that varies based on the cuda/hip version but can be overridden by the user.

- This PR is a bugfix
- It does the following:
  - Fixes #1194 #1195

### CODE REVIEW

**Comment by MrBurmark** [comment]
@mclarsen Does this fix your issue?

**Comment by mclarsen** [comment]
I think it does. Thanks for fixing it!

**Comment by rhornung67** [comment]
@MrBurmark before merging this, please make sure @davidbeckingsale reviews and approves. This somewhat disrupts his recent work to clean up our CMake variables.

**Review by davidbeckingsale — APPROVED** [review]
This looks fine to me

File: cmake/SetupRajaOptions.cmake [hunk 1]
```diff
@@ -17,8 +17,6 @@ option(RAJA_ENABLE_ROCTX "Build with ENABLE_ROCTX support" Off)
 option(RAJA_ENABLE_TBB "Build TBB support" Off)
 option(RAJA_ENABLE_TARGET_OPENMP "Build OpenMP on target device support" Off)
 option(RAJA_ENABLE_SYCL "Build SYCL support" Off)
-option(RAJA_ENABLE_EXTERNAL_CUB "Use an external cub" Off)
```
Comment (rhornung67) [inline cmake/SetupRajaOptions.cmake:20]: @davidbeckingsale the changes to this file are mostly what I was concerned about. You're good with these changes

Comment (davidbeckingsale) [reply cmake/SetupRajaOptions.cmake:20]: Yes, due to ordering they had to be moved from here, and the changes @MrBurmark made still allow for the user to override any of the new config.

**Review by rhornung67 — APPROVED** [review]

**Comment by MrBurmark** [comment]
Ok, I'm going to merge this then. Thanks @davidbeckingsale.

**Pull request merged**